src/OrleansRuntime/GrainTypeManager/GrainTypeManager.cs

```diff
@@ -153,7 +155,15 @@ private void InitializeGrainClassData(SiloAssemblyLoader loader, bool strict)
 
         private void InitializeInvokerMap(SiloAssemblyLoader loader, bool strict)
         {
-            IEnumerable<KeyValuePair<int, Type>> types = loader.GetGrainMethodInvokerTypes(strict);
+            // Load all existing implementations.
+            var existing = new HashSet<int>(loader.GetGrainMethodInvokerTypes(strict).Select(_ => _.Key));
+            
+            // Create implementations for all types which do not already have an implementation.
+            GrainMethodInvokerGenerator.CreateForCurrentlyLoadedAssemblies(type => !existing.Contains(type));
```

Comment:
The problem is that `SiloAssemblyLoader` will throw if multiple `IGrainMethodInvoker`s exist for the same interface:
```c#
if (result.ContainsKey(ifaceId))
                throw new InvalidOperationException(string.Format("Grain method invoker classes {0} and {1} use the same interface id {2}", result[ifaceId].FullName, type.FullName, ifaceId));
```

I could remove that assumption and replace these two methods with a single call. What do you think?

I'm now calling the `IGrainMethodInvoker` generator from `GrainTypeManager.InitializeInvokerMap` and calling the `GrainReference` generator from `GrainTypeManager.InitializeGrainClassData`. Does that sound more reasonable?